packages/lib/src/vis/matrix/StickyGrid.tsx

```diff
@@ -1,39 +1,58 @@
-import { useContext, useState } from 'react';
-import type { ReactNode, Ref } from 'react';
+import type { ReactNode } from 'react';
+import { useContext, forwardRef } from 'react';
 
-import AnchorCell from './AnchorCell';
-import { GridSettingsContext } from './GridSettingsContext';
-import IndexTrack from './IndexTrack';
+import HeaderCells from './HeaderCells';
 import styles from './MatrixVis.module.css';
+import { SettingsContext, RenderedItemsContext } from './context';
 
 interface Props {
   children: ReactNode;
   style: React.CSSProperties;
 }
 
-function StickyGrid(props: Props, ref: Ref<HTMLDivElement>) {
+const StickyGrid = forwardRef<HTMLDivElement, Props>((props, ref) => {
   const { children, style } = props;
-  const { rowCount, columnCount } = useContext(GridSettingsContext);
+  const { rowCount, columnCount, cellSize, sticky } =
+    useContext(SettingsContext);
 
-  const [sticky, setSticky] = useState(true);
+  const renderedItems = useContext(RenderedItemsContext);
+  const {
+    overscanColumnStartIndex: colStart = 0,
+    overscanColumnStopIndex: colStop = 0,
+    overscanRowStartIndex: rowStart = 0,
+    overscanRowStopIndex: rowStop = 0,
+  } = renderedItems || {};
```

Comment:
This is where I make use of `renderedItems`. Note that I'm using the `overscan` indices instead of the `visible` inidices which are also available. This allows the same overscan UX improvement to apply to header cells.